build.ts

```diff
@@ -0,0 +1,78 @@
+import { build } from 'vite';
+import * as configs from './vite.configs';
+import { mkdir } from 'fs/promises';
+import { getBrowser } from 'scripts/util';
+import colorLog from 'scripts/log';
+import { exec } from 'child_process';
+import { emptydirSync } from 'fs-extra';
+
+/**
+ * Try to make a directory, dont error if the directory exists
+ *
+ * @param str - path to directory to try to make
+ */
+async function trymkdir(str: string) {
+	try {
+		await mkdir(str);
+	} catch (err) {
+		if (
+			!err ||
+			typeof err !== 'object' ||
+			!('code' in err) ||
+			err.code !== 'EEXIST'
+		) {
+			throw new Error('Something went wrong making build folder');
+		}
+	}
+}
+
+async function createDistributable() {
```

Comment:
```
import archiver from 'archiver';
import fs from 'fs';
import path from 'path';

async function createDistributable() {
	const curDir = path.resolve(path.dirname(''));
	const outputFile = `${curDir}/web-scrobbler-${process.env.BROWSER}.zip`;
	const output = fs.createWriteStream(outputFile);

	const archive = archiver('zip', {
		zlib: { level: 0 },
	});

	archive.pipe(output);

	archive.directory(`build/${process.env.BROWSER}`, false);
	archive.finalize();
}
```

I guess something like this would work  using the `archiver` package